module/VuFind/src/VuFind/Db/Service/UserResourceService.php

```diff
@@ -127,4 +138,107 @@ public function getStatistics(): array
         $stats = current($query->getResult());
         return $stats;
     }
+
+    /**
+     * Unlink rows for the specified resource. This will also automatically remove
+     * any tags associated with the relationship.
+     *
+     * @param User|int          $user        ID of user removing links
+     * @param string|array|null $resource_id ID (or array of IDs) of resource(s) to
+     *                                       unlink (null for ALL matching
+     *                                       resources)
+     * @param UserList|null     $list        list to unlink (null for ALL matching lists, with the destruction
+     *                                       of all tags associated with the $resource_id value; true for ALL
+     *                                       matching lists, but retaining any tags associated with the
+     *                                       $resource_id independently of lists)
+     *
+     * @return void
+     */
+    public function destroyLinks($user, $resource_id, $list = null)
```

Comment:
Would it make sense to default resource_id to null here, so that you can use named arguments when passing only a user and a list? I think that would make the code significantly more readable.